lib/Common/Common/MathUtil.h

```diff
@@ -44,17 +44,8 @@ class Math
     }
 
     static bool     FitsInDWord(int32 value) { return true; }
-    static bool     FitsInDWord(int64 value) { return ((int64)(signed int)(value & 0xFFFFFFFF) == value); }
-    static bool     FitsInDWord(size_t value)
-    {
-#if TARGET_32
-        CompileAssert(sizeof(size_t) == 4);
-        return FitsInDWord((int32) value);
-#elif TARGET_64
-        CompileAssert(sizeof(size_t) == 8);
-        return FitsInDWord((int64)value);
-#endif
-    }
+    static bool     FitsInDWord(size_t value) { return ((size_t)(signed int)(value & 0xFFFFFFFF) == value); }
+    static bool     FitsInDWord(int64 value) { return FitsInDWord((size_t)value); }
```

Comment:
humm right...
Should I reimplement it like the size_t version aka
`return ((int64)(signed int)(value & 0xFFFFFFFF) == value);`?